File: check_n_minted_tokens.js
const { Web3 } = require("web3");
const fs = require("fs");

const w3 = new Web3("http://localhost:8545");
const account = w3.eth.accounts[0];
w3.eth.defaultAccount = account;

const contract_info = JSON.parse(fs.readFileSync("./CONTRACT_DEPLOYMENT_INFO.json").toString());
const contract_abi = contract_info["CONTRACT_ABI"];
const contract_address = contract_info["CONTRACT_ADDRESS"];

const contract = new w3.eth.Contract(contract_abi, contract_address);

async function interact() {
    const name = await contract.methods.name().call();
    console.log(name);
}

interact();

File: check_n_minted_tokens.py
"""
"""

import json

from web3 import Web3


if __name__ == "__main__":
    with open("./CONTRACT_DEPLOYMENT_INFO.json", "r") as f:
        token_info = json.loads(f.read())


    w3 = Web3(Web3.IPCProvider("./chain/execution/data/geth.ipc"))
    account = w3.eth.accounts[0]

    w3.eth.default_account = account

    contract = w3.eth.contract(
        address=token_info["CONTRACT_ADDRESS"],
        abi=token_info["CONTRACT_ABI"],
    )

    # name = str(contract.functions.greet().call())
    # print(name)

    n_tokens = str(contract.functions.totalSupply().call())
    print(n_tokens)


